cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -969,19 +997,23 @@ data ChainInformation = ChainInformation
   , slotInEpoch :: Word64
     -- ^ Relative slot number of the tip of the current chain within the
     -- epoch.
+  , blocksLost :: Int64
+    -- ^ Blocks lost due to switch at fork
```

Comment:
If you do end up following my overall proposal, then this comment is wrong and also the name is misleading.

I'd call it `myBlocksSelected`, an the comment would say something along the lines of: it's the difference between the numbers of my blocks on the latest selection minus that same number as of the last restart. Thus comparing `myBlocksSelected` to the number of blocks I've forged since the last restart shows how many blocks I've forged for naught.

Reply:
Looks like that ^^^ plan would require an additional metric that simply just increments with each `TraceForgedBlock` event.

Reply:
I've updated the code to what I think you mean.

Reply:
OK, after a little more thinking, I actually think one metric will suffice, and it can be "blocks forged since last restart but not currently selected".

`TraceForgedBlock` should increment the metric. `SwitchedToAFork` and `AddedToCurrentChain` should both increment it for each block of ours on the old chain and also decrement it for each block of ours on the new chain.

If we restart and then immediately switch to a fork with more of ours blocks, then this metric might dip negative. But over time, we should generally see this metric gradually increase.

Does that make sense? I'm headed to bed now -- talk to you tomorrow.

Reply:
I feel like `TraceForgedBlock` shouldn't increment the metric because when we forge a block it is appended to the current chain and is therefore not one of the unselected ones?

Reply:
The `TraceForgedBlock` just means we created the block and are about to add it to the ChainDB. Once it's added, then the ChainDB might select it, which would incur a separate `AddedToCurrentChain` event. So decrement for the forge, then increment for the subsequent selection, thus giving us the desired delta of 0.

Reply:
Done.  I've used the term "uncoupled" to mean blocks forged not on the current chain, which is the thing being counted.